src/physics/exact_solutions/exact_solution.cpp

```diff
@@ -0,0 +1,95 @@
+#include <deal.II/base/function.h>
+#include "exact_solution.h"
+
+namespace PHiLiP {
+
+// ========================================================
+// ZERO -- Returns zero everywhere; used a placeholder when no exact solution is defined.
+// ========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction_Zero<dim,nstate,real>
+::ExactSolutionFunction_Zero()
+        : ExactSolutionFunction<dim,nstate,real>()
+{
+    t = 0;//time_compare;
+}
+
+template <int dim, int nstate, typename real>
+inline real ExactSolutionFunction_Zero<dim,nstate,real>
+::value(const dealii::Point<dim,real> &/*point*/, const unsigned int /*istate*/) const
+{
+    //NEED TO UPDATE THIS
+    //(void) point;
+    real value = 0;
+    return value;
+}
+
+// ========================================================
+// 1D SINE -- Exact solution for advection_explicit_time_study
+// ========================================================
+template <int dim, int nstate, typename real>
+ExactSolutionFunction_1DSine<dim,nstate,real>
+::ExactSolutionFunction_1DSine (double time_compare)
+        : ExactSolutionFunction<dim,nstate,real>()
+{
+    t = time_compare;
+}
+
+template <int dim, int nstate, typename real>
+inline real ExactSolutionFunction_1DSine<dim,nstate,real>
+::value(const dealii::Point<dim,real> &point, const unsigned int /*istate*/) const
```

Comment:
I think `t` should be made an argument here rather than being fixed, and in your integration test you set `time_compare` and pass that there.